transport_analysis/analysis/velocityautocorr.py

```diff
@@ -5,111 +5,146 @@
 This module contains the :class:`VelocityAutocorr` class.
 
 """
-from typing import Union, TYPE_CHECKING
+from typing import TYPE_CHECKING
 
 from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.core.groups import UpdatingAtomGroup
 import numpy as np
 
 if TYPE_CHECKING:
-    from MDAnalysis.core.universe import Universe, AtomGroup
+    from MDAnalysis.core.universe import AtomGroup
 
 
 class VelocityAutocorr(AnalysisBase):
-    """VelocityAutocorr class.
-
-    This class is used to perform analysis on a trajectory.
+    """
+    Class to calculate a velocity autocorrelation function (VACF).
 
     Parameters
     ----------
-    universe_or_atomgroup: :class:`~MDAnalysis.core.universe.Universe` or :class:`~MDAnalysis.core.groups.AtomGroup`
-        Universe or group of atoms to apply this analysis to.
-        If a trajectory is associated with the atoms,
-        then the computation iterates over the trajectory.
-    select: str
-        Selection string for atoms to extract from the input Universe or
-        AtomGroup
+    atomgroup : AtomGroup
+        An MDAnalysis :class:`~MDAnalysis.core.groups.AtomGroup`.
+        Note that :class:`UpdatingAtomGroup` instances are not accepted.
+    dim_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
```

Comment:
You could probably make it so that `yx` is ok, I think it would still make logical sense in that the dims don't have an order except by convention, would still follow the `xy` code-path

Reply:
So let it accept all possible permutations? Sure, that would be good.